pulser-core/pulser/devices/_device_datacls.py

```diff
@@ -579,6 +579,140 @@ def to_abstract_repr(self) -> str:
+            (
+                " - Minimum distance between neighbouring atoms: "
+                f"{self.min_atom_distance} μm"
+            ),
+            f" - Maximum layout filling fraction: {self.max_layout_filling}",
+            f" - SLM Mask: {'Yes' if self.supports_slm_mask else 'No'}",
+        ]
+
+        if self.max_sequence_duration is not None:
+            lines.append(
+                " - Maximum sequence duration: "
+                f"{self.max_sequence_duration} ns"
+            )
+
+        device_lines = [
+            "\nDevice parameters:",
+        ]
+        if self.max_runs is not None:
+            device_lines.append(f" - Maximum number of runs: {self.max_runs}")
+        device_lines += [
+            (
+                " - Channels can be reused: " "Yes"
+                if self.reusable_channels
+                else "No"
+            ),
+            f" - Supported bases: {', '.join(self.supported_bases)}",
+            f" - Supported states: {', '.join(self.supported_states)}",
+        ]
+        if self.interaction_coeff is not None:
+            device_lines.append(
+                f" - Ising interaction coefficient: {self.interaction_coeff}"
+            )
+        if self.interaction_coeff_xy is not None:
+            device_lines.append(
+                f" - XY interaction coefficient: {self.interaction_coeff_xy}"
+            )
+
+        if self.default_noise_model is not None:
+            device_lines.append(
+                f" - Default noise model: {self.default_noise_model}"
+            )
+
+        layout_lines = [
+            "\nLayout parameters:",
+            f" - Requires layout: {'Yes' if self.requires_layout else 'No'}",
+        ]
+        if hasattr(self, "accepts_new_layouts"):
+            layout_lines.append(
+                " - Accepts new layout: " "Yes"
+                if self.accepts_new_layouts
+                else "No"
+            )
+
+        layout_lines += [
+            f" - Minimal number of traps: {self.min_layout_traps}",
+            f" - Maximal number of traps: {self.max_layout_traps}",
+        ]
+
+        ch_lines = ["\nChannels:"]
+        for name, ch in {**self.channels, **self.dmm_channels}.items():
+            if for_docs:
+                max_amp = "None"
+                if ch.max_abs_detuning is not None:
+                    max_amp = f"{float(cast(float, ch.max_amp)):.4g} rad/µs"
+
+                max_abs_detuning = "None"
+                if ch.max_abs_detuning is not None:
+                    max_abs_detuning = (
+                        f"{float(cast(float, ch.max_abs_detuning)):.4g} rad/µs"
+                    )
+
+                bottom_detuning = "None"
+                if hasattr(ch, "bottom_detuning"):
+                    if ch.bottom_detuning is not None:
+                        bottom_detuning = (
+                            f"{float(cast(float, ch.bottom_detuning)):.4g}"
+                            " rad/µs"
```

Comment:
Did you try this?
```suggestion
                if isinstance(ch, DMM) and ch.bottom_detuning is not None:
                    bottom_detuning = (
                        f"{float(ch.bottom_detuning):.4g} rad/µs"
```

Reply:
I have modified as you suggested. I also removed the cast for  `max_abs_detuning` because mypy was also complaining locally. However, I did not remove it for `max_map` because then I get an error:
```
pulser-core/pulser/devices/_device_datacls.py:664: error: Argument 1 to "float" has incompatible type "float | None"; expected "str | Buffer | SupportsFloat | SupportsIndex"  [arg-type]
```

Let's see what mypy gives now.